http/src/main/java/io/micronaut/http/filter/FilterRunner.java

```diff
@@ -552,8 +551,8 @@ private static FilterReturnHandler prepareReturnHandler(ConversionService conver
                     return next.handle(context, null, continuation);
                 }
 
-                Mono<?> publisher = Mono.from(Publishers.convertPublisher(conversionService, returnValue, Publisher.class));
-
+                Mono<?> monoPublisher = Mono.from(Publishers.convertPublisher(conversionService, returnValue, Publisher.class));
+                Publisher<?> publisher = ReactivePropagation.propagate(context.propagatedContext, monoPublisher);
```

Comment:
I think the Mono.from should happen after the propagate call

Reply:
@dstepanov can you address this?

Reply:
yea im not sure. ReactivePropagation.propagate has some logic that branches on Flux vs Mono, which after this PR might hit the different branch. But I doubt there's any impact.